### PR TITLE
AIOD-281

### PR DESCRIPTION
**Pull request by n4hm3**
https://crick.atlassian.net/browse/AIOD-281 - Implement cellpose fine-tuning
https://crick.atlassian.net/browse/AIOD-251 - Reviewed nxf widget refactor

changes after last review:
remove … in abstract classes
channels parameter
add input params: weight decay lr SGD momentum
Add test data option
indicate to save model to reg in UI
some utility function for saving fine-tuned models

### CODE REVIEW

File: src/ai_on_demand/widget_classes.py [hunk 1]
```diff
@@ -125,6 +140,25 @@ def store_config(self, save_dir, config_name):
...
+    def get_manifests(self):
+        # Re-retrieve manifests, including cache directory if available
+        cache_dir = None
+        nxf_widget = self.subwidgets.get("nxf")
+        if nxf_widget:
+            cache_dir = getattr(nxf_widget, "nxf_store_dir", None)
+        self.all_manifests = load_manifests(
+            filter_access=True, cache_dir=cache_dir
+        )
+        self.subwidgets[
```
Comment (n4hm3) [inline src/ai_on_demand/widget_classes.py]: Add a guard here to check if the model subwdiget exists
